### PR TITLE
Validation of board via hwdef configuration

### PR DESCRIPTION
**Pull request by bugobliterator**
Add feature in hwdef to be able to create board specific validation tests. Starts with replacing CubeBlack specific test with one written in the CubeBlack hwdef.

### CODE REVIEW

**Comment by proficnc** [comment]
Love it!

**Comment by bugobliterator** [comment]
> Shouldn't this PR also remove the "#define HAL_CHIBIOS_ARCH_CUBEBLACK" macro ? Is it still needed ?

Yes, you are right, I will remove it. Thanks for noticing.

File: libraries/AP_BoardConfig/board_drivers.cpp [hunk 1]
```diff
@@ -262,39 +260,19 @@ void AP_BoardConfig::validate_board_type(void)
...
-
 /*
   auto-detect board type
  */
 void AP_BoardConfig::board_autodetect(void)
 {
-#if defined(HAL_CHIBIOS_ARCH_CUBEBLACK)
-    check_cubeblack();
-    return;
+#if defined(VALIDATE_BOARD)
```
Comment (tridge) [inline libraries/AP_BoardConfig/board_drivers.cpp]: I'd prefer the macro to be HAL_VALIDATE_BOARD, just to keep the namespaces a bit separate

Comment (bugobliterator) [reply libraries/AP_BoardConfig/board_drivers.cpp]: > I like the concept, thanks Sid!
> I've suggested a small macro name change, plus can you please run:
> ./Tools/scripts/configure_all.py --no-bl --build
> and check that all boards still build ok? Then check bootloader builds.
> After structural changes to hwdef processing I like to check all builds

ok, I will check the builds, I thought the CI tests cover that, but I will double check.

**Pull request merged**